pysrc/bytewax/connectors/kafka/registry.py

```diff
@@ -0,0 +1,222 @@
+"""TODO."""
+import logging
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import Dict, Optional
+
+import requests
+from confluent_kafka.schema_registry import SchemaRegistryClient
+
+from .serde import (
+    SchemaDeserializer,
+    SchemaSerializer,
+    _AvroDeserializer,
+    _AvroSerializer,
+    _ConfluentAvroDeserializer,
+    _ConfluentAvroSerializer,
+)
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class SchemaConf:
+    """Info used to retrieve a schema from a schema registry.
+
+    Either use `schema_id`, or specify the `subject` and optionally `version`.
+    If no `version` is specified, it defaults to the latest schema.
+    """
+
+    # TODO: Only `avro` supported for now, but we might want
+    #       to add `protobuf` and others too
+    # format: str = "avro"
+    schema_id: Optional[int] = None
+    subject: Optional[str] = None
+    version: Optional[int] = None
```

Comment:
I didn't want to end up with a `Optional[Union[int, SchemaRef]]]` type on the argument, my idea was that you either specify the schema or not. But I agree with the other comment that having to pass `None` in the registry to avoid (de)serialization is not exactly a nice api, and this'll probably feel better if we move this to its own operator